internal/pkg/agent/application/upgrade/artifact/download/snapshot/downloader.go

```diff
@@ -131,11 +131,9 @@ func snapshotURI(versionOverride string, config *artifact.Config) (string, error
 		}
 
 		index := strings.Index(uri, "/beats/elastic-agent/")
-		if index == -1 {
-			return "", fmt.Errorf("not an agent uri: '%s'", uri)
+		if index != -1 {
```

Comment:
Can you add a comment explaining why this would happen, or maybe better add a small unit test to document what the possible problems here are (the need to handle unexpected agent components)?

Reply:
I agree that a wee explanation why an earlier return ins't appropriated could be good

Reply:
I thought about writing a unittest, but it depends on the iteration order on a map, hence even with the bug it will randomly pass :/ I could run the test some hundred times, but still not deterministic.

Anyways, I'll add a comment explaining.

Reply:
Done on 4a7c1b8a7